src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -76,6 +71,11 @@ internal class TracerProviderSdk : TracerProvider
                 {
                     OpenTelemetrySdkEventSource.Log.ActivityStarted(activity);
 
+                    if (this.needInstrumentations && legacyActivityOperationNames.Contains(activity.OperationName))
```

Comment:
if the activity is created using `ActivitySource` and it *happens* to have same operation name as one of the legacyActivityOperationNames, then we'd be running sampler for them again here, which would be incorrect. We need to do this only for legacy activities - i.e the ones which has ActivitySource empty.

Reply:
Yeah this is something I missed. There could be cases where ActivitySource creates an activity with the same operation name as the one created by DiagnosticSource. I have added a check for empty ActivitySource.